packages/synapse-react-client/src/components/SynapseTable/TopLevelControls.tsx

```diff
@@ -126,6 +128,9 @@ const TopLevelControls = (props: TopLevelControlsProps) => {
   }
 
   const refresh = () => {
+    // clear selection
+    setSelectedRows([])
+    // refresh the data
     executeQueryRequest(getLastQueryRequest())
```

Comment:
This does not seem execute the query (on CRC researcher).  This is an existing bug, so I'm ignoring for now, but may want to track down what is happening here (do we need to invalidate the queries for react-query before this?)

Reply:
I don't think we'll need to fix this for the Cohort Builder either, since it will switch views rather that refresh the current page data.

Reply:
I'm not that familiar with this, but I would guess that you're right about it being a query invalidation issue. Is this invoked after mutating the table? The re-fetch would probably just result in a cache hit if the update call doesn't invalidate the cache (which it probably doesn't, since this hasn't been closely looked at since before we started using react-query).

Can you open a Jira ticket with some repro steps?

Reply:
Opened https://sagebionetworks.jira.com/browse/PORTALS-2682